apps/emqx/src/emqx_ds_schema.erl

```diff
@@ -0,0 +1,266 @@
+                importance => ?IMPORTANCE_MEDIUM,
+                desc => ?DESC(messages)
+            })}
+    ].
+
+fields(builtin) ->
+    %% Schema for the builtin backend:
+    [
+        {backend,
+            sc(
+                builtin,
+                #{
+                    'readOnly' => true,
+                    default => builtin,
+                    importance => ?IMPORTANCE_MEDIUM,
+                    desc => ?DESC(builtin_backend)
+                }
+            )},
+        {'_config_handler',
+            sc(
+                {module(), atom()},
+                #{
+                    'readOnly' => true,
+                    default => {?MODULE, translate_builtin},
+                    importance => ?IMPORTANCE_HIDDEN
+                }
+            )},
+        {data_dir,
+            sc(
+                string(),
+                #{
+                    mapping => "emqx_durable_storage.db_data_dir",
+                    required => false,
+                    importance => ?IMPORTANCE_MEDIUM,
+                    desc => ?DESC(builtin_data_dir)
+                }
+            )},
+        {n_shards,
+            sc(
+                pos_integer(),
+                #{
+                    default => 16,
+                    importance => ?IMPORTANCE_MEDIUM,
+                    desc => ?DESC(builtin_n_shards)
+                }
+            )},
+        {replication_factor,
+            sc(
+                pos_integer(),
+                #{
+                    default => 3,
+                    importance => ?IMPORTANCE_HIDDEN
+                }
+            )},
+        {local_write_buffer,
+            sc(
+                ref(builtin_local_write_buffer),
+                #{
+                    importance => ?IMPORTANCE_MEDIUM,
+                    desc => ?DESC(builtin_local_write_buffer)
+                }
+            )},
+        {layout,
+            sc(
+                hoconsc:union(builtin_layouts()),
+                #{
+                    desc => ?DESC(builtin_layout),
+                    importance => ?IMPORTANCE_MEDIUM,
+                    default =>
+                        #{
+                            <<"type">> => wildcard_optimized
+                        }
+                }
+            )}
+    ];
+fields(builtin_local_write_buffer) ->
+    [
+        {max_items,
+            sc(
+                pos_integer(),
+                #{
+                    default => 1000,
+                    mapping => "emqx_durable_storage.egress_batch_size",
+                    importance => ?IMPORTANCE_MEDIUM,
+                    desc => ?DESC(builtin_local_write_buffer_max_items)
+                }
+            )},
+        {flush_interval,
+            sc(
+                emqx_schema:timeout_duration_ms(),
+                #{
+                    default => 100,
+                    mapping => "emqx_durable_storage.egress_flush_interval",
+                    importance => ?IMPORTANCE_MEDIUM,
```

Comment:
I'd propose to leave that one hidden for now, since current egress implementation will likely need to undergo a rework in the future in my impression.

Reply:
Done